src/components/hack-details.js

```diff
@@ -36,66 +36,31 @@ const useStyles = makeStyles(theme => ({
 export function HackDetails(props) {
   const classes = useStyles();
 
-  const [data] = useState();
-  const [title, setTitle] = useState("");
-  const [description, setDescription] = useState("");
-  const [team, setTeam] = useState();
-  const [goal, setGoal] = useState("");
+  const [data, setData] = useState(null);
 
   useEffect(() => {
     //console.log(props.match.params);
 
     if (props.match !== undefined) {
       Axios.get(
-        process.env.REACT_APP_API_URL + "hackdetail/" + props.match.params.id
+        UrlJoin(process.env.REACT_APP_API_URL,"hackdetail/",props.match.params.id)
       ).then(res => {
-        setTitle(res.data.title);
-        setDescription(res.data.description);
-        setGoal(res.data.goal);
-        setTeam(res.data.team);
+        setData(res.data)
       });
     }
-  }, [data, props.match]);
+  }, [props.match]);
 
   const joinHack = event => {
```

Comment:
you can probably just push this function down into the `JoinButton` component, and just pass down the hackId as the prop, the `JoinButton` can connect the the store to access the `userId`